### PR TITLE
{tools}[foss/2018b] SYMPHONY v5.6.16

### PR DESCRIPTION
**Pull request by tstrempel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by zao** [comment]
Test report by @zao
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
freja - Linux ubuntu 18.04, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.17
See https://gist.github.com/b57428643f2aaf68e2de4addb6215cb3 for a full test report.

File: easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+toolchain = {'name': 'foss', 'version': '2018b'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://www.coin-or.org/download/source/SYMPHONY/']
+sources = [SOURCE_TGZ]
+checksums = ['54df3fd953d9a2e4178f10373c6cb8b46a0808569b42adf07aa6d0044b2ab04b']
+
+sanity_check_paths = {
+    'files': ['bin/symphony', 'lib/libsym.%s' % SHLIB_EXT, ],
+    'dirs': ['bin', 'lib', 'includei/coin'],
```
Comment (zao) [inline easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb]:
```suggestion
    'dirs': ['bin', 'lib', 'include/coin'],
```

**Comment by zao** [comment]
Test report by @zao
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
freja - Linux ubuntu 18.04, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.17
See https://gist.github.com/5aa155fc06a4f29cd9706952cd2e3ba4 for a full test report.

**Comment by tstrempel** [comment]
@tstrempel I updated the dependencies. It should work now.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 6 out of 6 (1 easyconfigs in this PR)
node104.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/0a2b122cf9f69f04bfb4a2a1d98b2ee9 for a full test report.

File: easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+sources = [SOURCE_TGZ]
+checksums = ['54df3fd953d9a2e4178f10373c6cb8b46a0808569b42adf07aa6d0044b2ab04b']
+
+dependencies = {
+    ('CoinUtils', '2.11.3'),
+    ('Osi', '0.108.5'),
+    ('Clp', '1.17.3'),
+    ('Cgl', '0.60.2'),
+}
+
```
Comment (lexming) [inline easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb:22]: As I said, the dependencies set in the easyconfig seem to not be used during the build of SYMPHONY. Please add the following `configopts` to ensure that it uses them
```suggestion

# Use BLAS/LAPACK from OpenBLAS
configopts = '--with-blas="$LIBBLAS" --with-lapack="$LIBLAPACK" '
# Disable GLPK, dependencies have to be built with it as well
configopts += '--without-glpk '
# Use CoinUtils from EB
configopts += '--with-coinutils-lib="-lCoinUtils" --with-coinutils-incdir=$EBROOTCOINUTILS/include/coin '
configopts += '--with-coinutils-datadir=$EBROOTCOINUTILS/share/coin/Data'
# Use Clp from EB
configopts += '--with-clp-lib="-lOsiClp -lClpSolver -lClp" --with-clp-incdir="$EBROOTCLP/include/coin" '
configopts += '--with-clp-datadir=$EBROOTCLP/share/coin/Data '
# Use Osi from EB (also needs links to Clp due to OsiClpSolver)
configopts += '--with-osi-lib="-lOsiClp -lClpSolver -lClp -lOsi" '
configopts += '--with-osi-incdir="$EBROOTOSI/include/coin -I$EBROOTCLP/include/coin" '
configopts += '--with-osi-datadir=$EBROOTOSI/share/coin/Data '
# Use Cgl from EB
configopts += '--with-cgl-lib="-lCgl" --with-cgl-incdir="$EBROOTCGL/include/coin" '
configopts += '--with-cgl-datadir=$EBROOTCGL/share/coin/Data '
```

File: easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+sanity_check_paths = {
+    'files': ['bin/symphony', 'lib/libSym.%s' % SHLIB_EXT],
+    'dirs': ['bin', 'lib', 'include/coin'],
```
Comment (lexming) [inline easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb:25]: Sanity checks should be updated as well to cover more files in the installation
```suggestion
    'files': ['bin/symphony', 'lib/libSym.%s' % SHLIB_EXT, 'lib/libOsiSym.%s' % SHLIB_EXT],
    'dirs': ['include/coin', 'lib/pkgconfig', 'share/coin']
```

**Comment by tstrempel** [comment]
@lexming Thanks for the suggestions, I thought it would autodetect the dependencies, so I will update my pr.

File: easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+description = "SYMPHONY is an open-source solver for mixed-integer linear programs (MILPs) written in C."
+
+toolchain = {'name': 'foss', 'version': '2018b'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://www.coin-or.org/download/source/SYMPHONY/']
+sources = [SOURCE_TGZ]
+checksums = ['54df3fd953d9a2e4178f10373c6cb8b46a0808569b42adf07aa6d0044b2ab04b']
+
+builddependencies = {
```
Comment (lexming) [inline easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb:16]:
```suggestion
builddependencies = [
```

File: easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    ('Autotools', '20180311'),
+    ('Doxygen', '1.8.14'),
+    ('pkg-config', '0.29.2'),
+}
```
Comment (lexming) [inline easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb:20]:
```suggestion
]

```

File: easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+dependencies = {
```
Comment (lexming) [inline easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb:21]:
```suggestion
dependencies = [
```

File: easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    ('CoinUtils', '2.11.3'),
+    ('Osi', '0.108.5'),
+    ('Clp', '1.17.3'),
+    ('Cgl', '0.60.2'),
+}
```
Comment (lexming) [inline easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb:26]:
```suggestion
]
```

File: easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+configopts += '--with-clp-lib="-lOsiClp -lClpSolver -lClp" --with-clp-incdir="$EBROOTCLP/include/coin" '
+configopts += '--with-clp-datadir=$EBROOTCLP/share/coin/Data '
+# Use Osi from EB (also needs links to Clp due to OsiClpSolver)
+configopts += '--with-osi-lib="-lOsiClp -lClpSolver -lClp -lOsi" '
+configopts += '--with-osi-incdir="$EBROOTOSI/include/coin -I$EBROOTCLP/include/coin" '
+configopts += '--with-osi-datadir=$EBROOTOSI/share/coin/Data '
+# Use Cgl from EB
+configopts += '--with-cgl-lib="-lCgl" --with-cgl-incdir="$EBROOTCGL/include/coin" '
+configopts += '--with-cgl-datadir=$EBROOTCGL/share/coin/Data '
+sanity_check_paths = {
```
Comment (lexming) [inline easybuild/easyconfigs/s/SYMPHONY/SYMPHONY-5.6.16-foss-2018b.eb:45]:
```suggestion

sanity_check_paths = {
```

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 2 out of 3 (1 easyconfigs in this PR)
node376.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/db3c03465b10468b5fb84b18788699d7 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node154.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/effcaec3b983be53e94925cce7f5e30a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, x86_64, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/1613f5f86a1198d145881aa9e91a7a23 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3149.skitty.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/03bbd82f83f434f86a58da6e12fd3a7f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2501.golett.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/625254204eee4da8b859ac0ed993d03a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 11 out of 11 (1 easyconfigs in this PR)
node3401.kirlia.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/149573b870dad59662cccf60daa573c5 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @tstrempel!

**Pull request merged**